assets/src/models/createDetourMachine.ts

```diff
@@ -68,6 +76,23 @@ export const createDetourMachine = setup({
       }
     }),
 
+    "fetch-detour-directions": fromPromise<
+      Awaited<ReturnType<typeof fetchDetourDirections>>,
+      {
+        points?: ShapePoint[]
+      }
+    >(async ({ input: { points } }) => {
+      if (points) {
+        if (points.length < 2) {
+          return Ok({ coordinates: [], directions: undefined })
+        } else {
+          return fetchDetourDirections(points)
+        }
+      } else {
+        throw "Missing detour direction inputs"
+      }
```

Comment:
nit/suggestion(**_non-blocking_**): thoughts? 😅 I kinda wanna do this to `fetch-route-patterns` now too(later).

I don't wanna blow up your PR, so I've formatted this and tested this so that you could use `commit-suggestion`, but don't feel like you have to.

```suggestion
      if (!points) {
        throw "Missing detour direction inputs"
      }
      if (points.length < 2) {
        return Ok({ coordinates: [], directions: undefined })
      }
      return fetchDetourDirections(points)
 
```